uint32 current_stage

string task_state